test/extended/dr/static_pod_rollouts.go

```diff
@@ -0,0 +1,145 @@
+package dr
+
+import (
+	"fmt"
+	"time"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	exutil "github.com/openshift/origin/test/extended/util"
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var _ = g.Describe("[sig-etcd][Feature:DisasterRecovery][Suite:openshift/etcd/recovery][Disruptive] etcd", func() {
+	defer g.GinkgoRecover()
+	oc := exutil.NewCLIWithoutNamespace("etcd-static-pod-rollouts").AsAdmin()
+	etcdNamespace := "openshift-etcd"
+
+	g.BeforeEach(func() {
+		isSingleNode, err := exutil.IsSingleNode(context.Background(), oc.AdminConfigClient())
+		o.Expect(err).NotTo(o.HaveOccurred())
+		if isSingleNode {
+			g.Skip("the test is for etcd peer communication which is not valid for single node")
+		}
+	})
+
+	g.AfterEach(func(ctx context.Context) {
+
+		// Reset the hardware speed back to default to leave the cluster in the same state.
+		g.GinkgoT().Log("resetting hardware speed back to default")
+		data := fmt.Sprintf(`{"spec": {"controlPlaneHardwareSpeed": ""}}`)
+		_, err := oc.AdminOperatorClient().OperatorV1().Etcds().Patch(ctx, "cluster", types.MergePatchType, []byte(data), metav1.PatchOptions{})
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		// select the same node from which the etcd instance was removed to add it back
+		masterNodes := masterNodes(oc)
+		etcdTargetNode := masterNodes[0]
+
+		g.GinkgoT().Log("debugging into the same node to add back the etcd static pod manifest")
+		err = oc.AsAdmin().Run("debug").Args("node/"+etcdTargetNode.Name, "--", "chroot", "/host", "/bin/bash", "-c", "mv /var/lib/etcd-backup/etcd-pod.yaml /etc/kubernetes/manifests/  && rm -r /var/lib/etcd-backup").Execute()
```

Comment:
nice trick! you can leave out the removal at the end, I think it's fine to keep some yaml files sitting around